### ISSUE TITLE
Add Period Filter to Virtual Cards in Collective Settings

### ISSUE THREAD

**Issue by SudharakaP** [opencollective/opencollective#4270]
We need to add a period filter to the virtual cards section in collective settings. This was initially proposed in the figma design; https://www.figma.com/file/ZQBMWhnGGtRWeIZknFW1eP/%5BOC-Design%5D-Production-Ready-%E2%9C%85?node-id=10720%3A177694 related to the issue, https://github.com/opencollective/opencollective/issues/4199. But we had some doubts about how it should behave and thus was not included in the initial iteration. The related slack discussion is; 

https://opencollective.slack.com/archives/C0RMV6F8C/p1619108700150800

The behaviour as per Amina's comment is; 

>The Period drop down will have the options of ‘Last week’ ‘Last month’ ‘Last 3 months’ ‘Last 6 months’ and ‘Last year’
Basically we filter based on the period the card was created in!

**Comment by Betree** [comment]
Don't we want something that can pick using date ranges, like suggested in https://github.com/opencollective/opencollective/issues/4070?

**Comment by Betree** [comment]
@SudharakaP In https://github.com/opencollective/opencollective-frontend/pull/6345, I'm changing the `PeriodFilter` component to use date pickers rather than intervals (6 months, 2 years, etc). Feel free to copy the old component if you want to implement such behavior.

**Comment by aminakazi** [comment]
Love the date picker approach. Let's go ahead with it.
Looking at it live now, I think we'll need some kind of an empty state for when there are no transactions/no virtual cards in that particular timeline. I'll share that with you @SudharakaP and @Betree 

@SudharakaP We should also restrict the user from going beyond the date on which the virtual card feature had been announced. 

@Memo-Es From a UX POV, Do you reckon we need some kind of a loader while the period resets?

**Comment by Betree** [comment]
@aminakazi There is an empty state for transactions, it looks like that:
![image](https://user-images.githubusercontent.com/1556356/117854693-07249100-b28a-11eb-93e2-6ad6825c57e8.png)
